### PR TITLE
[bsdiff_classic] Update Bzip2 compat

### PR DESCRIPTION
**Pull request by nhz2**
This will help testing https://github.com/JuliaIO/BSDiff.jl/pull/49

### CODE REVIEW

File: B/bsdiff_classic/build_tarballs.jl [hunk 1]
```diff
@@ -3,11 +3,11 @@
 using BinaryBuilder, Pkg
 
 name = "bsdiff_classic"
-version = v"4.3.0"
```
Comment (nhz2) [inline B/bsdiff_classic/build_tarballs.jl:6]: Does this need to be updated or should I leave it as 4.3.0?

File: B/bsdiff_classic/build_tarballs.jl [hunk 2]
```diff
@@ -31,10 +31,8 @@ products = [
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    # Future versions of bzip2 should allow a more relaxed compat because the
-    # soname of the macOS library shouldn't change at every patch release.
-    Dependency("Bzip2_jll", v"1.0.6"; compat="=1.0.6"),
+    Dependency("Bzip2_jll", v"1.0.8"; compat="1.0.8"),
```
Comment (giordano) [inline B/bsdiff_classic/build_tarballs.jl:34]:
```suggestion
    Dependency("Bzip2_jll"; compat="1.0.8"),
```

File: B/bsdiff_classic/build_tarballs.jl (hunk 1 above)
Comment (giordano) [reply B/bsdiff_classic/build_tarballs.jl:6]: You do need to change the version number if compat bounds change (and you changed also the julia one below)

**Review by giordano — COMMENTED** [review]
You may want to trigger a rebuild of Bzip2 in a separate PR, so that we get it built for aarch64-freebsd (hopefully that works out-of-the-box without much hassle)

**Comment by giordano** [comment]
License file is missing

**Comment by nhz2** [comment]
> License file is missing

Nice catch. 
I don't see a specific license file, so I'm adding the two ".c" files with a license at the top.

File: B/bsdiff_classic/build_tarballs.jl [hunk 3]
```diff
@@ -17,6 +17,7 @@ perl -i -ple '$_ = "#include <sys/types.h>\n" . $_ if $. == 31' bspatch.c
 cc -O3 -lbz2 -I"${prefix}/include" bsdiff.c -o bsdiff
 cc -O3 -lbz2 -I"${prefix}/include" bspatch.c -o bspatch
 install bsdiff bspatch "${bindir}"
+install_license bsdiff.c bspatch.c
```
Comment (giordano) [inline B/bsdiff_classic/build_tarballs.jl:20]: Using source code as license file isn't a great idea.  That seems to be 2-clause BSD, we have that somewhere in the rootfs

Comment (giordano) [reply B/bsdiff_classic/build_tarballs.jl:20]: It should be
```suggestion
install_license /usr/share/licenses/BSD2
```

Comment (nhz2) [reply B/bsdiff_classic/build_tarballs.jl]: That file is missing the name and year

Comment (nhz2) [reply B/bsdiff_classic/build_tarballs.jl]: I updated the script to add the top lines of the c file as a license with the correct name and year.

Comment (giordano) [reply B/bsdiff_classic/build_tarballs.jl]: I'm not particularly enthusiastic about cutting files around, what if in a next version the file slightly change (e.g. they add a new line at the top of the file)?  No one will tell you that the license file is now truncated.  We ship the license files precisely for these cases.

Comment (nhz2) [reply B/bsdiff_classic/build_tarballs.jl]: Yes, but how should I replace the "Copyright (c) [year], [fullname]" in "/usr/share/licenses/BSD2" with "Copyright 2003-2005 Colin Percival"?

**Comment by fingolfin** [comment]
> This will help testing [JuliaIO/BSDiff.jl#49](https://github.com/JuliaIO/BSDiff.jl/pull/49)

That PR is titled "Bump compat for CodecBzip2, TranscodingStreams, and SuffixArrays" so how exactly does changing this JLL help with testing that PR?

**Comment by nhz2** [comment]
BSDiff.jl uses CodecBzip2.jl and bsdiff_classic_jll for testing, but the latest version of CodecBzip2.jl requires Bzip2_jll 1.0.8 or later, while bsdiff_classic_jll currently needs exactly Bzip2_jll 1.0.6

**Comment by nhz2** [comment]
I added patches and copyright info from https://salsa.debian.org/debian/bsdiff/-/tree/debian/4.3-17

This is the latest version from Debian I can figure out how to compile. 4.3-18 includes some mmap code that doesn't compile.

**Pull request merged**